### PR TITLE
Updating flexible fleet configuration

### PR DESCRIPTION
**Pull request by JoeJimFlood**
## Proposed changes

This pull request adds calibration constants to tour mode choice for zero-auto households using flexible fleet services, as well as fixes a few miscellaneous issues in model configs relating to flexible fleets.

## Impact

Flexible fleet should be used more now by people living in zero-auto households and less by people who have vehicles available. The base year number of boardings should be about the same, but the model sensitivity to auto ownership should increase.

## Types of changes

What types of changes does your code introduce to ABM?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] Updates were tested in just the resident model and calibrated
- [x] A full model run (Scenario ID 216) was done and the results were checked to make sure they were close to the target.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Further comments

An analysis of trips made by members of zero-auto households within flexible fleet service areas found that significantly more travelers were choosing DRIVEALONE as a trip mode over TNC_SHARED. This seemed unrealistic, so new coefficients were added to tour mode choice to increase the TNC_SHARED tours of travelers from zero-auto households.

To calculate a target, data from the 2022 household travel survey was analyzed (T:\Survey\HHTS\Wave_1_Deliverables). Trips made within Downtown San Diego (census tracts 51.01, 51.02, 52.01, 52.02, 53.01, 53.02, 54.01, 54.02, 54.03, 56.01, 58.01, and 58.02) by people with zero cars were isolated, and the percentage of these trips that used pooled TNC was calculated (1.3%). The calibration coefficients were increased until the TNC_SHARED share for tours within the Downtown/Little Italy NEV service area made by people from zero-auto households matched that target. `coef_calib_flexfleet` was consequentially lowered to make sure that the base year NEV boardings matched their target.

Other issues in the configurations identified by @aber-sandag were addressed.

### CODE REVIEW

File: src/asim/configs/crossborder/trip_mode_choice.csv [hunk 1]
```diff
@@ -66,7 +66,7 @@ util_TNC_SINGLE_IVT,TNC Single - In-vehicle time,c_ivt * s2_time_skims,,,,,,,,,1
 util_TNC_SINGLE_wait,TNC Single - Wait time,c_ivt * 1.5 * tnc_single_wait_time,,,,,,,,,1,
 util_TNC_SINGLE_cost,TNC Single - Cost,@c_cost * ((TNC_single_baseFare + (df.s2_dist_skims * TNC_single_costPerMile) + (df.s2_time_skims * TNC_single_costPerMinute).clip(lower=TNC_single_costMinimum)) * 100 + df.s2_cost_skims),,,,,,,,,1,
 util_TNC Shared_switch,TNC Shared - switch turn-off (depends on data availability),@((~df.nev_available) & (~df.microtransit_available) & (scenarioYear==2022)),,,,,,,,,,-999
-util_TNC_SHARED_IVT,TNC Shared - In-vehicle time,"@df.c_ivt * np.where(df.nev_available, df.nev_time, np.where(df.microtransit_available, df.microtransit_time, df.s3_time_skims)) * tnc_shared_ivt_factor",,,,,,,,,,1
+util_TNC_SHARED_IVT,TNC Shared - In-vehicle time,"@df.c_ivt * np.where(df.nev_available, df.nev_time, np.where(df.microtransit_available, df.microtransit_time, df.s3_time_skims)) * TNC_shared_IVTFactor",,,,,,,,,,1
```
Comment (aber-sandag) [inline src/asim/configs/crossborder/trip_mode_choice.csv:69]: TNC_shared_IVTFactor should only apply to non-ff trips, should be moved inside paratheses as with resident model. Same for visitor tour and trip mode choice

**Review by aber-sandag — CHANGES_REQUESTED** [review]
TNC_shared_IVTFactor was moved in resident mode choice to only apply to non-ff TNC_SHARED trips. Same change should be made in crossborder and visitor

File: src/asim/configs/common_airport/write_trip_matrices_annotate_trips_preprocessor.csv [hunk 2]
```diff
@@ -211,7 +211,7 @@ Description,Target,Expression
 #,_timeDrive,"_timeDrive + (odt_skims['HOV2_M_TIME'] + _TAXI_WAIT_TIME) * np.where((trip_mode == 'TAXI'),1,0)"
 #,_timeDrive,"_timeDrive + (odt_skims['HOV2_M_TIME'] + _SINGLE_TNC_WAIT_TIME) * np.where((trip_mode == 'TNC_SINGLE'),1,0)"
 ,_timeDrive,"_timeDrive + odt_skims['HOV3_M_TIME'] * np.where(((trip_mode == 'SHARED3') & vot2),1,0)"
-#,_timeDrive,"_timeDrive + (odt_skims['HOV3_M_TIME'] + _SHARED_TNC_WAIT_TIME) * np.where((trip_mode == 'TNC_SHARED'),1,0) * _TNC_SHARED_IVT_FACTOR"
+#,_timeDrive,"_timeDrive + (odt_skims['HOV3_M_TIME'] + _SHARED_TNC_WAIT_TIME) * TNC_shared_IVTFactor * np.where((trip_mode == 'TNC_SHARED'),1,0) * _TNC_SHARED_IVT_FACTOR"
```
Comment (bhargavasana) [inline src/asim/configs/common_airport/write_trip_matrices_annotate_trips_preprocessor.csv:214]: Please check if _TNC_SHARED_IVT_FACTOR is still needed.

Comment (aber-sandag) [reply src/asim/configs/common_airport/write_trip_matrices_annotate_trips_preprocessor.csv:214]: _TNC_SHARED_IVT_FACTOR no longer exists, this line is commented out

Comment (aber-sandag) [reply src/asim/configs/common_airport/write_trip_matrices_annotate_trips_preprocessor.csv:214]: No TNC_SHARED mode in airport model

Comment (aber-sandag) [reply src/asim/configs/common_airport/write_trip_matrices_annotate_trips_preprocessor.csv]: Removed these commented out lines to avoid confusion

**Review by bhargavasana — APPROVED** [review]

**Review by aber-sandag — APPROVED** [review]

**Pull request merged**